code/modules/reagents/reagent_containers/food/drinks/drinkingglass.dm

```diff
@@ -15,571 +15,57 @@
-			if(FUEL)
-				icon_state = "dr_gibb_glass"
-				name = "glass of welder fuel"
-				desc = "Unless you are an industrial tool, this is probably not safe for consumption."
-			if(BROWNSTAR)
-				icon_state = "brownstar"
-				item_state = "brownstar"
-				name = "\improper Brown Star"
-				desc = "It's not what it sounds like..."
-			if(ICETEA)
-				icon_state = "icedteaglass"
-				item_state = "icedteaglass"
-				name = "iced tea"
-				desc = "Like tea, but refreshes rather than relaxes."
-			if(ARNOLDPALMER)
-				icon_state = "arnoldpalmer"
-				name = "\improper Arnold Palmer"
-				desc = "Known as half and half to some. A mix of ice tea and lemonade."
-			if(MILKSHAKE)
-				icon_state = "milkshake"
-				item_state = "milkshake"
-				name = "milkshake"
-				desc = "Brings all the boys to the yard."
-			if(LEMONADE)
-				icon_state = "lemonade"
-				item_state = "lemonade"
-				name = "lemonade"
-				desc = "Oh, the nostalgia..."
-			if(KIRASPECIAL)
-				icon_state = "kiraspecial"
-				name = "\improper Kira Special"
-				desc = "Long live the guy who everyone had mistaken for a girl. Baka!"
-			if(REWRITER)
-				icon_state = "rewriter"
-				name = "\improper Rewriter"
-				desc = "This will cure your dyslexia and cause your arrhythmia."
-			if(PINACOLADA)
-				icon_state = "pinacolada"
-				name = "\improper Pina Colada"
-				desc = "If you like this and getting caught in the rain, come with me and escape."
-			if(SANGRIA)
-				icon_state = "sangria"
-				item_state = "sangria"
-				name = "\improper Sangria"
-				desc = "So sweet you won't notice the alcohol until you're wasted."
-			if(DANS_WHISKEY)
-				icon_state = "dans_whiskey"
-				item_state = "dans_whiskey"
-				name = "\improper Discount Dan's 'Malt' Whiskey"
-				desc = "The cheapest path to liver failure."
-			if(GREYVODKA)
-				icon_state = "ginvodkaglass"
-				item_state = "ginvodkaglass"
-				name = "glass of greyshirt vodka"
-				desc = "A questionable concoction of ingredients found within maintenance. Tastes just like you'd expect."
+
+			name = R.glass_name ? R.glass_name : "glass of " + R.name //uses glass of [reagent name] if a glass name isn't defined
+			desc = R.glass_desc ? R.glass_desc : R.description //uses the description if a glass description isn't defined
+			isGlass = R.glass_isGlass
+
+			if(R.glass_icon_state)
+				icon_state = R.glass_icon_state
+				item_state = R.glass_icon_state
+			else
+				icon_state ="glass_colour"
+				item_state ="glass_colour"
+				var/image/filling = image('icons/obj/reagentfillings.dmi', src, "glass")
+				filling.icon += mix_color_from_reagents(reagents.reagent_list)
+				filling.alpha = mix_alpha_from_reagents(reagents.reagent_list)
+				overlays += filling
+	else
+		icon_state = "glass_empty"
+		item_state = "glass_empty"
+		name = "drinking glass"
+		desc = "Your standard drinking glass."
+
+	if(iscarbon(loc))
+		var/mob/living/carbon/M = loc
+		M.update_inv_hands()
+
+		/*
 			if(PINTPOINTER)
```

Comment:
If that is commented out already, you can just remove it.

Reply:
good catch